src/queues/EventRedactionQueue.ts

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019-2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { LogLevel, MatrixClient } from "matrix-bot-sdk"
+import { ERROR_KIND_FATAL } from "../ErrorCache";
+import { logMessage } from "../LogProxy";
+import { RoomUpdateError } from "../models/RoomUpdateError";
+import { redactUserMessagesIn } from "../utils";
+
+export interface QueuedRedaction {
+    /** The room which the redaction will take place in. */
+    readonly roomId: string;
+    /**
+     * Carries out the redaction task and is called by the EventRedactionQueue
```

Comment:
Nit:
```js
/**
 * Carry out the redaction.
 * 
 * Called by the EventRedactionQueue.
 *
 * @param ...
 */
```

Short sentences are usually better :)

Also, what does this return?

Reply:
I'm not sure it's supposed to return anything, I will change to `Promise<void>`